### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ajgoyalnoiro**

### CODE REVIEW

**Comment by tbachman** [comment]
Please fix the PEP8 issues.

File: aim/api/types.py [hunk 1]
```diff
@@ -29,11 +29,13 @@
...
 icmpv4_type = {'0': 'echo-rep', '3': 'dst-unreach',
                '4': 'src-quench', '8': 'echo',
-               '11': 'time-exceeded', '255': 'unspecified'}
+               '11': 'time-exceeded', '255': UNSPECIFIED}
+icmpv4_code = {'0': 'no-code', '0xffff': UNSPECIFIED}
 
 icmpv6_type = {'0': UNSPECIFIED, '1': 'dst-unreach', '3': 'time-exceeded',
                '128': 'echo-req', '129': 'echo-rep', '135': 'nbr-solicit',
                '136': 'nbr-advert', '137': 'redirect'}
+icmpv6_code = {'0xffff': UNSPECIFIED}
```
[the remark this answers is not in the document]
Comment (tbachman) [reply aim/api/types.py]: These are what are defined in the ACI model.

[the remark this answers is not in the document]
Comment (kentwu1) [reply aim/api/types.py]: sure

[the remark this answers is not in the document]
Comment (ajgoyalnoiro) [reply aim/api/types.py]: These codes are per IANA and the ACI modes

File: aim/agent/aid/universes/aci/converter.py [hunk 2]
```diff
@@ -764,6 +766,10 @@ def bgp_as_id_converter(object_dict, otype, helper, to_aim=True):
                        'converter': port},
             'ethertype': {'other': 'ethertype',
                           'converter': ethertype},
+            'icmpType': {'other': 'icmp_type',
+                         'converter': icmpv4_type},
```
[the remark this answers is not in the document]
Comment (ajgoyalnoiro) [reply aim/agent/aid/universes/aci/converter.py]: Yes, the aci model only says 'icmpType and icmpCode'

File: aim/db/migration/alembic_migrations/versions/308d91b3dadf_security_group_rule_icmp_code.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+
+    op.add_column(
+        'aim_security_group_rules',
+        sa.Column('icmp_type', sa.String(16), nullable=False,
+                  server_default='255')
```
[the remark this answers is not in the document]
Comment (ajgoyalnoiro) [reply aim/db/migration/alembic_migrations/versions/308d91b3dadf_security_group_rule_icmp_code.py]: as the resource is being initialized with t.UNSPECIFIED, modified this too

**Comment by kentwu1** [comment]
Pls test it out on a real fab before we merge this pull request then.

**Pull request merged**